R/defineModel.R

```diff
@@ -0,0 +1,65 @@
+#' Defining the Model
+#'
+#' Give the complete formula and generate the expected kernel library.
+#'
+#' It processes data based on formula and label_names and creates a kernel
+#' library according to the parameters given in Kern_par.
+#'
+#' @param formula A symbolic description of the model to be fitted.
+#' @param label_names A character string indicating all the interior variables
+#' included in each predictor.
+#' @param data A dataframe to be fitted.
+#' @param Kern_par A dataframe indicating the parameters of base kernels to be
+#' created.
+#' @return \item{Y}{Reponses of the dataframe.}
+#'
+#' \item{X1}{The first type of factor in the dataframe (could contains several
+#' subfactors).}
+#'
+#' \item{X2}{The second type of factor in the dataframe (could contains several
+#' subfactors).}
+#'
+#' \item{Kernlist}{The kernel library containing several kernels given by
```

Comment:
make sure to stress this is a list of kernel functions. (otherwise user may think this is a list of kernel matrices)